SOURCES/0002-backport.patch

```diff
@@ -0,0 +1,360 @@
+@@ -217,6 +217,7 @@
+ 	struct timecounter tc;
+ 	struct timespec64 prev_ptp_time; /* Pre-reset PTP clock */
+ 	ktime_t ptp_reset_start; /* Reset time in clock mono */
++	char fw_version[32];
+ };
+ 
+ void igc_up(struct igc_adapter *adapter);
+--- a/igc_base.c	2022-09-23 12:17:01.000000000 +0000
++++ b/igc_base.c	2022-09-24 07:07:24.414598890 +0000
+@@ -205,6 +205,7 @@
+ 	case IGC_DEV_ID_I220_V:
+ 	case IGC_DEV_ID_I225_K:
+ 	case IGC_DEV_ID_I225_K2:
++	case IGC_DEV_ID_I226_K:
+ 	case IGC_DEV_ID_I225_LMVP:
+ 	case IGC_DEV_ID_I225_IT:
+ 	case IGC_DEV_ID_I226_LM:
+--- a/igc_defines.h	2022-09-23 12:17:01.000000000 +0000
++++ b/igc-module/igc_defines.h	2022-09-24 06:29:34.767001177 +0000
+@@ -160,6 +160,7 @@
+ #define IGC_NVM_RW_REG_START	1    /* Start operation */
+ #define IGC_NVM_RW_ADDR_SHIFT	2    /* Shift to the address bits */
+ #define IGC_NVM_POLL_READ	0    /* Flag for polling for read complete */
++#define IGC_NVM_DEV_STARTER	5    /* Dev_starter Version */
+ 
+ /* NVM Word Offsets */
+ #define NVM_CHECKSUM_REG		0x003F
+--- a/igc_ethtool.c	2022-09-23 12:17:01.000000000 +0000
++++ b/igc_ethtool.c	2022-09-24 06:43:08.841321756 +0000
+@@ -129,10 +129,27 @@
+ 				    struct ethtool_drvinfo *drvinfo)
+ {
+ 	struct igc_adapter *adapter = netdev_priv(netdev);
++	struct igc_hw *hw = &adapter->hw;
++	u16 nvm_version = 0;
++	u16 gphy_version;
+ 
+ 	strlcpy(drvinfo->driver,  igc_driver_name, sizeof(drvinfo->driver));
+ 
+-	/* add fw_version here */
++	/* NVM image version is reported as firmware version for i225 device */
++	hw->nvm.ops.read(hw, IGC_NVM_DEV_STARTER, 1, &nvm_version);
++
++	/* gPHY firmware version is reported as PHY FW version */
++	gphy_version = igc_read_phy_fw_version(hw);
++
++	scnprintf(adapter->fw_version,
++			sizeof(adapter->fw_version),
++			"%x:%x",
++			nvm_version,
++			gphy_version);
++
++	strscpy(drvinfo->fw_version, adapter->fw_version,
++		sizeof(drvinfo->fw_version));
++
+ 	strlcpy(drvinfo->bus_info, pci_name(adapter->pdev),
+ 		sizeof(drvinfo->bus_info));
+ 
+@@ -1018,11 +1035,11 @@
+ 	switch (cmd->flow_type) {
+ 	case TCP_V4_FLOW:
+ 		cmd->data |= RXH_L4_B_0_1 | RXH_L4_B_2_3;
+-		fallthrough;
++//		fallthrough;
+ 	case UDP_V4_FLOW:
+ 		if (adapter->flags & IGC_FLAG_RSS_FIELD_IPV4_UDP)
+ 			cmd->data |= RXH_L4_B_0_1 | RXH_L4_B_2_3;
+-		fallthrough;
++//		fallthrough;
+ 	case SCTP_V4_FLOW:
+ 	case AH_ESP_V4_FLOW:
+ 	case AH_V4_FLOW:
+@@ -1032,11 +1049,11 @@
+ 		break;
+ 	case TCP_V6_FLOW:
+ 		cmd->data |= RXH_L4_B_0_1 | RXH_L4_B_2_3;
+-		fallthrough;
++//		fallthrough;
+ 	case UDP_V6_FLOW:
+ 		if (adapter->flags & IGC_FLAG_RSS_FIELD_IPV6_UDP)
+ 			cmd->data |= RXH_L4_B_0_1 | RXH_L4_B_2_3;
+-		fallthrough;
++//		fallthrough;
+ 	case SCTP_V6_FLOW:
+ 	case AH_ESP_V6_FLOW:
+ 	case AH_V6_FLOW:
+@@ -1899,7 +1916,7 @@
+ }
+ 
+ static const struct ethtool_ops igc_ethtool_ops = {
+-	.supported_coalesce_params = ETHTOOL_COALESCE_USECS,
++//	.supported_coalesce_params = ETHTOOL_COALESCE_USECS,
```

Comment:
Is this something not supported by kernel 4.19?

Reply:
Not supported in 4.19